docs/api/js/utils.md

```diff
@@ -1,68 +1,785 @@
-    DEPOSIT: 600_000,
-    EXECUTE: 620_000,
-    ERC20_APPROVE: 50_000
-};
+```ts
+export async function getERC20BridgeCalldata(
+    l1TokenAddress: string,
+    l1Sender: string,
+    l2Receiver: string,
+    amount: BigNumberish,
+    provider: ethers.providers.Provider
+): Promise<string> {
+    const gettersData = await getERC20GettersData(l1TokenAddress, provider);
+    return L2_BRIDGE_ABI.encodeFunctionData('finalizeDeposit', [
+        l1Sender,
+        l2Receiver,
+        l1TokenAddress,
+        amount,
+        gettersData
+    ]);
+}
 ```
+
+### `getERC20GettersData`
+
+Gets the data needed for initializing an L1 token counterpart on L2.
+
+#### Inputs
+
+- `l1TokenAddress`: token address on L1 as string.
+- `provider`: ethers `Provider` object.
+
+#### Outputs
+
+An array of:
+
+- `nameBytes`: `bytes` object representation of token name.
+- `symbolBytes`: `bytes` object representation of token symbol.
+- `decimalBytes`: `bytes` object representation of token decimal representation.
+
+```ts
+async function getERC20GettersData(l1TokenAddress: string, provider: ethers.providers.Provider): Promise<string> {
+    const token = IERC20MetadataFactory.connect(l1TokenAddress, provider);
+
+    const name = await token.name();
+    const symbol = await token.symbol();
+    const decimals = await token.decimals();
+
+    const coder = new AbiCoder();
+
+    const nameBytes = coder.encode(['string'], [name]);
+    const symbolBytes = coder.encode(['string'], [symbol]);
+    const decimalsBytes = coder.encode(['uint256'], [decimals]);
+
+    return coder.encode(['bytes', 'bytes', 'bytes'], [nameBytes, symbolBytes, decimalsBytes]);
+}
+```
+
+### `getL2HashFromPriorityOp`
+
+Returns the hash of the L2 priority operation from a given transaction receipt and L2 address.
+
+#### Inputs
+
+- `txReceipt`: ethers `TransactionReceipt` object.
+- `zkSyncAddress`: address of sender/transaction ?? on L2 as `Address` object.
+
+```ts
+export function getL2HashFromPriorityOp(
+    txReceipt: ethers.providers.TransactionReceipt,
+    zkSyncAddress: Address
+): string {
+    let txHash: string = null;
+    for (const log of txReceipt.logs) {
+        if (log.address.toLowerCase() != zkSyncAddress.toLowerCase()) {
+            continue;
+        }
+
+        try {
+            const priorityQueueLog = ZKSYNC_MAIN_ABI.parseLog(log);
+            if (priorityQueueLog && priorityQueueLog.args.txHash != null) {
+                txHash = priorityQueueLog.args.txHash;
+            }
+        } catch {}
+    }
+    if (!txHash) {
+        throw new Error('Failed to parse tx logs');
+    }
+
+    return txHash;
+}
+```
+
+### `getHashedL2ToL1Msg`
+
+Returns a keccak encoded message with a given sender address and block number from the L1 messenger contract.
+
+#### Inputs
+
+- `sender`: sender address on L1 ?? as `Address` object.
+- `msg`: encoded message as `BytesLike` object.
+- `txNumberInBlock`: block number containing message on L2 ?? as number.
```

Comment:
it is the index of the transaction in block

Reply:
added